scanner/src/main/kotlin/utils/FileListResolver.kt

```diff
@@ -74,18 +74,14 @@ private fun ProvenanceFileStorage.getFileList(provenance: KnownProvenance): File
     return yaml.fromYaml()
 }
 
-private val IGNORED_DIRECTORY_MATCHER by lazy {
-    FileMatcher(patterns = VCS_DIRECTORIES.map { "**/$it" })
-}
-
 private fun createFileList(dir: File): FileList {
     val files = dir.walk().onEnter {
-        !IGNORED_DIRECTORY_MATCHER.matches(it.relativeTo(dir).invariantSeparatorsPath)
+        it.name !in VCS_DIRECTORIES
     }.filter {
         it.isFile
     }.mapTo(mutableSetOf()) {
         FileEntry(path = it.relativeTo(dir).invariantSeparatorsPath, sha1 = HashAlgorithm.SHA1.calculate(it))
     }
 
-    return FileList(ignorePatterns = IGNORED_DIRECTORY_MATCHER.patterns.toSet(), files = files)
+    return FileList(ignorePatterns = VCS_DIRECTORIES.mapTo(mutableSetOf()) { "**/$it" }, files = files)
```

Comment:
Previously, the patterns passed to file list have been used to filter things. So, there was no risk that the patterns passed to `FileList` is inconsistent with the patterns used for the filtering. 

The new code requires the lambda in line 78 - 80 to be consistent with the patterns passed in line 86. Therefore, I slightly prefer the previous code.

What do you think?

Reply:
I've dropped that commit.